### ISSUE TITLE
Can not store plugin name  in logs ( which is then used by  tensorboard to read data for relavant plugin)

### ISSUE THREAD

**Issue by sahilpasricja** [tensorflow/tensorflow#32130]
**System information**
- OS Platform and Distribution -  Linux Ubuntu 18.04:
- TensorFlow version (or github SHA if from source):1.14.0

In my existing model, i am using CNN model with tf.layers 

def mnist_cnn(inputs):

    

`
def mnist_cnn(inputs):

    input_layer = tf.reshape(inputs, [-1, 28, 28, 3])

    # Convolutional Layer #1
    conv1 = tf.layers.conv2d(
          inputs=input_layer,
          filters=32,
          kernel_size=[5, 5],
          padding="same",
          activation=tf.nn.relu,
          name="conv1",
    )
	  

    # Pooling Layer #1
    pool1 = tf.layers.max_pooling2d(inputs=conv1, pool_size=[2, 2], strides=2)

    # Convolutional Layer #2 and Pooling Layer #2
    conv2 = tf.layers.conv2d(
          inputs=pool1,
          filters=64,
          kernel_size=[5, 5],
          padding="same",
          activation=tf.nn.relu)

    # weights_1_array = np.zeros((786, 28, 28, 32))
    # conv_kernel_1 = tf.nn.conv2d(conv1, weights_1_array, [1, 4, 4, 1], padding='SAME')

    # kernel = tf.get_collection(tf.GraphKeys.VARIABLES, 'conv1/kernel')[0]

    pool2 = tf.layers.max_pooling2d(inputs=conv2, pool_size=[2, 2], strides=2)

    # Dense Layer
    pool2_flat = tf.reshape(pool2, [-1, 7 * 7 * 64])
    dense = tf.layers.dense(inputs=pool2_flat, units=1024, activation=tf.nn.relu)
    dropout = tf.layers.dropout(inputs=dense, rate=0.4)

    # Logits Layer
    outputs = tf.layers.dense(inputs=dropout, units=10)
`
And i used to write logs using summary 

` tensor_summaries = list(map(lambda graph_node:
                                tf.summary.tensor_summary(graph_node.name,
                                                          graph.get_tensor_by_name(graph_node.name + ":0"),
                                                          summary_metadata=tf.SummaryMetadata(
                                                              display_name=graph_node.name,
                                                              summary_description=description,
                                                              plugin_data=tf.SummaryMetadata.PluginData(
                                                                  plugin_name=PLUGIN_NAME)),
                                                          collections=collections),
                                tensor_ops))`

Within this summary, there was a feature to write plugin name , this plugin name is then used by my custom tensor board to identify data from my plugin.

With tf keras , we have callbacks like save.model to store weight but that don't give an option to write plugin name this makes it impossible for tensorboard to identify  data for my plugin , 

Is there any way to write logs with plugin name using call back ?

**Comment by nfelt** [comment]
So the new way to write arbitrary tensor summaries is using `tf.compat.v2.summary.write()`, which can accept a serialized summary metadata: https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/summary/write

cc @omalleyt12 - the TensorBoard keras callback works with custom summary ops now, right?  So I believe that `write()` would also work within the callback.

**Comment by sahilpasricja** [comment]
@nfelt , @omalleyt12  Thanks for your suggestions although we are looking in same direction but our visions are not converging. Let me try again.
Of course, there are ways to write logs from TensorFlow using write and tf.compat.v2.summary.write() but my problem is 
I was earlier using TF.layers
while run 
for every epoch --> summary_op = summary_function(default_graph)
sees.run(summary_op)

In summary function --> I was fetching the weights tensors and adding summary data using 
plugin_data=tf.SummaryMetadata.PluginData( plugin_name=PLUGIN_NAME)), 
and rrturn this to main function which then calls 
sess..run(summary_op)
writer.add_summary(summary, i)


NOW WITH KERAS
I have no way to add function directly with epochs, neither I can trigger sessions.
with callbacks, if I call the same summary function, it returns no data as the scope of session is within the test.fit function. 

Even if I trigger session manually, Keras still trigger it's on session my triggered session never fetches any data.

so callbacks are able to call the function but the summary function is designed to work with sessions and keras has it's on way to handling sessions

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this, as you are using an older version of tensorflow which is officially considered end of life . We recommend that you upgrade to the latest 2.x version and let us know if the issue still persists in newer versions. Please open a new issue for any help you need against 2.x, and we will get you the right help. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, please provide us with more information.
